docs/content/concepts/terminology.md

```diff
@@ -1,100 +1,192 @@
+
+As the kcp server is based on the Kubernetes API server, it provides very similar configuration options and uses etcd
+as its datastore, so if you have experience with operating Kubernetes, you can very easily apply it to kcp.
+
+At the end, the kcp server and all logical clusters created on top of it can be accessed using the regular Kubernetes
+tooling such as kubectl and client-go.
+
+!!! note
+    The only exception to this are Kubernetes controllers. Regular Kubernetes controllers can work with a single
+    (logical) cluster. However, if you're building a developer platform on top of kcp, you likely want your controller
+    to reconcile objects across multiple logical clusters. At the moment, this is not supported by upstream Kubernetes
+    controller-runtime. Instead you have to use
+    [our fork of it](https://github.com/kcp-dev/controller-runtime/tree/kcp-0.19/examples/kcp).
+    We're working with the upstream community on adding support for multi-cluster setups to the Kubernetes
+    controller-runtime.
+
 ## Logical Cluster
 
-A logical cluster is a way to subdivide a single kube-apiserver + etcd storage into multiple clusters (different APIs,
-separate semantics for access, policy, and control) without requiring multiple instances.  A logical cluster is a
-mechanism for achieving separation, but may be modelled differently in different use cases.  A logical cluster is
-similar to a virtual cluster as defined by sig-multicluster, but is able to amortize the cost of a new cluster to be
-zero or near-zero memory and storage so that we can create tens of millions of empty clusters cheaply.
+A logical cluster is a way to subdivide a kcp server and its `etcd` datastore into multiple clusters without requiring
+multiple API server and etcd instances. A logical cluster is able to amortize the cost of a new cluster to be near-zero
+memory and storage, so that we can create a huge number of empty clusters cheaply.
+
+Each logical cluster has it's own set of APIs installed, it's own objects, and different access semantics and
+policies, i.e. RBAC rules. In other words, each logical cluster is isolated from other logical clusters.
+
+This is accomplished by adding an additional attribute to an object's identifier in etcd and kcp server to associate
+object with its logical cluster. On the storage level, the regular Kubernetes API server identifies objects using
+(group, version, resource, optional namespace, name). The kcp server enriches this identifier with logical cluster
+name, so we have (group, version, resource, **logical cluster name**, optional namespace, name).
+
+A logical cluster provides Kubernetes-cluster-like HTTPS/CRUD endpoints, i.e. endpoints that typical Kubernetes client
+tooling (e.g. client-go, controller-runtime, and others) can interact with as they would with regular Kubernetes
+clusters. Using the appropriate kubeconfig file, a user can run `kubectl get all -A` to return all objects in all
```

Comment:
nit: is all really all? :-) I believe in Kube it is not.

Reply:
It definitely isn't. What about something like this?

> a user can run `kubectl get foo -A` to return all objects of type `foo` across all namespaces in that concrete logical cluster.